marlowe/security.md

```diff
@@ -0,0 +1,122 @@
+
+
+#### What else is occurring in the transaction?
+
+The unsigned transaction may contain other spending and payments beyond that specified for the Marlowe contract. This can be examined with the tool `cardano-cli transaction view`.
+
+
+### Marlowe Payout Validator
+
+In addition to the *semantics validator* discussed above, Marlowe has a *payout validator*. The same security considerations hold for it. Its script hash can be computed as follows:
+```bash
+nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- role validator --mainnet --out-file /dev/null --print-hash
+addr1w9yswm4tyqjrmj2xy5glhx9fe7m3n7rwj6fz3qfekly3mucd3rynq
+```
+```console
+addr1w9yswm4tyqjrmj2xy5glhx9fe7m3n7rwj6fz3qfekly3mucd3rynq
+Validator hash: "49076eab20243dc9462511fb98a9cfb719f86e9692288139b7c91df3"
+```
+
+Deserialization of the `Datum` and `Redeemer` should target `(Plutus.V2.Ledger.Api.CurrencySymbol, Plutus.V2.Ledger.Api.TokenName)` and `()`, respectively.
+
+
+## Role Tokens
+
+The Marlowe validator does not enforce [a particular monetary policy for role tokens](./best-practices.md#monetary-policy-for-role-tokens), in order to make possible novel use cases using role tokens. However, the security of authorizations in a role-based Marlowe contract depends critically upon the role-token monetary policy. Thus, one should carefully scrutinize both the monetary policy and the on-chain disbursement of the tokens before participating in a Marlowe contract. Verifying the monetary policy of a simple script just involves retrieving the script off of the blockchain and studying it; verifying the monetary policy of a Plutus script involves obtaining and studying the Plutus source code for the script and hashing the source code to check the monetary policy ID.
+
+One typically uses a monetary policy (like that supported in Marlowe Runtime) that enforces a single minting event and single tokens for each role: such a monetary policy ensures that the role tokens are true non-fungible tokens (NFTs) so that the holders of the role tokens are provably the only ones who can act as the parties in the contract.
+
+Nevertheless, monetary policies that mint multiple copies of a particular role token or ones with an open minting policy support non-standard use cases. For instance, minting two copies of each role token and distributing them to the same party allows that party to place one token in cold storage as a backup if their wallet containing the "hot" role token becomes inaccessible. Some novel crowd-source contracts might involve assigning the role (via identical role tokens that may be minted even after the contract commences) to many participants. Finally, a Plutus contract's minting policy for role tokens can coordinate with the operation of one or more Marlowe contracts.
```

Comment:
```suggestion
Nevertheless, monetary policies that mint multiple copies of a particular role token or ones with an open minting policy support non-standard use cases. For instance, minting two copies of each role token and distributing them to the same party allows that party to place one token in cold storage as a backup if their wallet containing the "hot" role token becomes inaccessible. Some novel crowd-sourcing contracts might involve assigning the role (via identical role tokens that may be minted even after the contract commences) to many participants. Finally, a Plutus contract's minting policy for role tokens can coordinate with the operation of one or more Marlowe contracts.
```
I am not 100% sure "crowd-source contract" is wrong, but in case it is a typo

Reply:
I think that either could be used, but https://idioms.thefreedictionary.com/crowd-source implies "crowdsourcing" is the more common usage.